node/options.go

```diff
@@ -138,8 +138,17 @@ func as(in interface{}, as interface{}) interface{} {
 
 	return reflect.MakeFunc(ctype, func(args []reflect.Value) (results []reflect.Value) {
 		outs := reflect.ValueOf(in).Call(args)
+
 		out := reflect.New(outType.Elem())
-		out.Elem().Set(outs[0])
+		if outs[0].Type().AssignableTo(outType.Elem()) {
```

Comment:
reason for this change?

Reply:
This allow providing fx.In structs as implementations of interfaces - in this case this is for `api.Common` needed by `advmgr.New`